packages/trpc/server/routers/viewer/auth.tsx

```diff
@@ -86,6 +87,24 @@ export const authRouter = router({
         throw new TRPCError({ code: "UNAUTHORIZED" });
       }
 
+      return;
+    }),
+  deploymentSetup: authedProcedure
+    .input(
+      z.object({
+        licenseType: z.nativeEnum(DeploymentLicenseType).optional(),
+        licenseKey: z.string().optional(),
+      })
+    )
+    .mutation(async ({ input }) => {
+      const data = {
+        licenseConsentAt: new Date(),
+        ...(input.licenseType ? { licenseType: input.licenseType } : {}),
+        ...(input.licenseKey ? { licenseKey: input.licenseKey } : {}),
+      };
+
+      await prisma.deployment.upsert({ where: { id: 1 }, create: data, update: data });
```

Comment:
Since this is a special kind of table we should abstract these methods to limit the where clauses.  Since there will be only one and will be confusing for newcomers. could be something like `deploymentSettingsUpdate/Update` etc.